src/fit.jl

```diff
@@ -1,5 +1,13 @@
 """
-    fit!(estimator, num_iterations, X, y[, test...]; [verbosity = 1, is_row_major = false])
```

Comment:
Just a comment here for reference. This change is just a removal  of an outdated doc signature for a method with iterations that's no longer used. The `num_iterations` has been removed from the `fit!` method in this [PR](https://github.com/IQVIA-ML/LightGBM.jl/pull/80/files) as it was not necessary (train! still has num_iterations). The `num_iterations` were initially introduced in the `fit!` but at the time it wasn't clear this was needed in the `fit!` method itself [reference](https://github.com/IQVIA-ML/LightGBM.jl/pull/51/files#r425872182).